jwst/superbias/tests/test_bias_sub.py

```diff
@@ -0,0 +1,216 @@
+       reference file during subtraction.'''
+
+    # Create inputs, subarray SUB320A335R data, and superbiases
+    ngroups = 5
+    nrows = 320
+    ncols = 320
+    blevel = 2000.
+    xstart = 486
+    ystart = 1508
+
+    data, bias = setup_subarray_cube(xstart, ystart, ngroups, nrows, ncols)
+    manualbias = bias.data[ystart-1:ystart-1+nrows, xstart-1:xstart-1+ncols]
+
+    # Add signal values and bias values
+    data.data[:] = blevel
+    manualbias[:] = blevel
+
+    # Run the pipeline
+    output = do_correction(data, bias)
+
+    # Make sure the subarray was extracted correctly from superbias file
+    # Make sure that manual superbias subtraction matches pipeline output
+    manualout = data.data - manualbias
+    assert np.array_equal(output.data, manualout)
+
+
+def test_dq_propagation(setup_full_cube):
+    '''Check that the PIXELDQ array of the science exposure is correctly
+       combined with the reference file DQ array.'''
+
+    # Create inputs, data, and superbiases
+    ngroups = 5
+    nrows = 2048
+    ncols = 2048
+    dqval1 = 5
+    dqval2 = 10
+
+    data, bias = setup_full_cube(ngroups, nrows, ncols)
+
+    # Add DQ values to the data and reference file
+    data.pixeldq[5, 5] = dqval1
+    bias.dq[5, 5] = dqval2
+
+    # Run the pipeline
+    output = do_correction(data, bias)
+
+    # Make sure DQ values from data and reference file are added in the output
+    assert output.pixeldq[5, 5] == dqval1 + dqval2
+
+
+def test_nans_in_superbias(setup_full_cube):
+    '''Check that no superbias subtraction is done for pixels that have a
+       value of NaN in the reference file.'''
+
+    # Create inputs, data, and superbiases
+    ngroups = 5
+    nrows = 2048
+    ncols = 2048
+    blevel = 2000.
+
+    data, bias = setup_full_cube(ngroups, nrows, ncols)
+
+    # Add signal values and bias values
+    data.data[:] = blevel
+    bias.data[:] = blevel
+
+    # Set superbias value for pixel to NaN
+    bias.data[500, 500] = np.nan
+
+    # Run the pipeline
+    output = do_correction(data, bias)
+
+    # Check that subtraction was not done on pixel with NaN reference value
+    # Check that subtraction was done on other pixels
+    assert np.array_equal(output.data[0, :, 500, 500], data.data[0, :, 500, 500])
+    assert np.array_equal(output.data[0, :, 50, 50], data.data[0, :, 50, 50] - blevel)
+
+
+def test_full_step(setup_full_cube):
+    '''Test full run of the SuperBiasStep.'''
+
+    # Create inputs, data, and superbiases
+    ngroups = 5
+    nrows = 2048
+    ncols = 2048
+    blevel = 2000.
+
+    data, bias = setup_full_cube(ngroups, nrows, ncols)
+
+    # Add signal values and bias values
+    data.data[0, :, 500, 500] = blevel
+
+    # Run the pipeline
+    output = SuperBiasStep.call(data)
+
+    # Check that expected reference file is used based on data headers
+    assert output.meta.ref_file.superbias.name == 'crds://jwst_nircam_superbias_0026.fits'
```

Comment:
Would it be better to use the mocked superbias reffile here?

Reply:
Given the check
```
assert output.data[0, 0, 500, 500] == pytest.approx(blevel - 2557.0452, 0.01)
```
One could I think do this easily with a mocked superbias reffile made from the `bias` already returned from `setup_full_cube`.  A way to do this would be to use pytest's `tmpdir` fixture.  Something like
```
def test_full_step(setup_full_cube, tmpdir):
    tmpfile = str(tmpdir.join('superbias_reffile.fits'))
    ...
    bias.save(tmpfile)
    output = SuperBiasStep.call(data, override_superbias=tmpfile)
    ...

Reply:
@jdavies-st After talking with @cracraft, we decided it was best to keep the tests as simple as possible and remove the reference file dependence altogether for superbias and saturation. The latest commit has the simplified test for the full superbias step.

Reply:
Looks great!  Many thanks.